### ISSUE TITLE
Export md file to PDF

### ISSUE THREAD

**Issue by lorenzoFabbri** [jupyterlab/jupyterlab#4676]
I have a very simple question. I wrote some reports in Markdown using JupyterLab. I'd like to export them to PDF now but I can't save theme from within JupyterLab. What are the options? Thank you.

**Comment by krassowski** [comment]
You could try https://github.com/aaren/notedown to convert your Markdown file into a jupyter notebook (from Jupyter Notebook) and then export to PDF from JupyterLab.

**Comment by ned2** [comment]
It would be great to see Jupyter Lab support this natively. None of the options in that Superuser thread are viable for the report I have (and I suspect this will be true for many others)

1. While I love grip, it's HTML output is not that great in this context. Our report uses <figure> elements for captioning images. Grip just rips the text of the figure out and dumps it raw between the previous and subsequent <p> elements. (whereas Jupyer Lab's Markdown previewer handles and formats figure elements perfectly).
2. pandoc does not support converting inline HTML in markdown into Latex/PDF, it just strips it out. See this [issue](https://github.com/jgm/pandoc/issues/2716).
3. I couldn't get markdown-pdf to install as the phantomjs dependency was breaking. As far as I'm aware, phantomjs is no longer maintained, so I don't think this is a good option anyway.

So basically, having the Jupyer Lab's Markdown preview be converted to PDF is I think the idea outcome here.

**Comment by blink1073** [comment]
Perhaps use marked to convert to html and then https://github.com/MrRio/jsPDF to render the pdf?

**Comment by blink1073** [comment]
This looks promising: https://github.com/MrRio/jsPDF/blob/2f51a5dc2a00b87762876729706a145f24566743/examples/html2pdf/showcase.html#L77

**Comment by ned2** [comment]
The solution I came up with based on @krassowski's suggestion:

    $ notedown --from markdown --to notebook report.md > report.ipynb
    $ jupyter nbconvert --to html report.ipynb
    $ google-chrome executive_summary.html
   
Then print to PDF with Chrome, disabling the headers/footers

Going straight to PDF directly with nbconvert is not an option, because this strips the HTML (maybe because it's using pandoc under the hood)

**Comment by gbrault** [comment]
Any progress? Would be awesome to have this natively supported by jupyter!

**Comment by phaustin** [comment]
Perhaps overkill for what you need, but it will become simpler as css3 page media support
arrives in browsers:

https://github.com/eoas-ubc/paged_html_theme#pagedjs-html-sphinx-theme
